src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/blob.py

```diff
@@ -0,0 +1,144 @@
+# ---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# ---------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+from datetime import datetime
+from azure.cli.command_modules.storage._command_type import cli_storage_data_plane_command
+from azure.cli.command_modules.storage._factory import blob_data_service_factory
+from azure.storage.blob import BlockBlobService
+from azure.storage.blob.models import Include
+
+
+def storage_blob_copy(client, source_account, source_container, destination_container,
+                      source_sas=None,
+                      pattern=None, recursive=False, snapshots=False,
+                      exclude_old=False, exclude_new=False, blob_type="block"):
+    """
+    Copy blobs between containers and storage accounts. This is a server-side copy operation
+    therefore the command is asynchronous.
+
+    :param str source_account:
+        The account name of the source storage account
+
+    :param str source_container:
+        The source blob container
+
+    :param str source_sas:
+        The shared access signature used to access the source container. It is not required if
+        either a connection string is given or the source container doesn't require a sas.
+
+    :param str destination_container:
+        The destination blob container
+
+    :param str pattern:
```

Comment:
Could you provide some example patterns (in the review, not the docstring) so we can see what patterns would correspond to different copy behaviors? I'm not sure why a --recursive option would be necessary and some examples might make that clear.